src/main/app/pay/payClient.ts

```diff
@@ -22,11 +22,19 @@ export class PayClient {
 
   async create (user: User,
                 pbaAccount: string,
-                caseReference: string,
+                externalId: string,
                 customerReference: string,
                 organisationName: string,
-                fee: FeeResponse): Promise<PaymentResponse> {
-    const paymentReq: object = this.preparePaymentRequest(pbaAccount, caseReference, customerReference, organisationName, fee)
+                fee: FeeResponse,
+                caseReference: string): Promise<PaymentResponse> {
```

Comment:
LGTM, although I would prefer to keep both externalId and caseReference close to each other.